engine/src/main/java/org/terasology/logic/console/ui/MiniChatOverlay.java

```diff
@@ -62,13 +65,21 @@ public void initialise() {
             @Override
             public String get() {
                 Iterable<Message> msgs = console.getMessages(CoreMessageType.CHAT, CoreMessageType.NOTIFICATION);
-                String last = "";
+                ArrayList<Message> msgsList = new ArrayList<>();
+                StringBuilder messageHistory = new StringBuilder();
 
                 for (Message msg : msgs) {
-                    last = msg.getMessage();
+                    msgsList.add(msg);
                 }
 
-                return last;
+                //checks if number of messages > MAX_LINES, else starts from 0
+                for (int i = msgsList.size() >= MAX_LINES ? msgsList.size() - MAX_LINES : 0; i < msgsList.size(); i++) {
+                    messageHistory.append(msgsList.get(i).getMessage());
+                    if (i < msgsList.size() - 1) {
+                        messageHistory.append(Console.NEW_LINE);
+                    }
+                }
+                return messageHistory.toString();
```

Comment:
That sounds preferable to copying the entire buffer first and pruning the copy to the last n messages afterwards.